hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/OmSnapshotManager.java

```diff
@@ -95,9 +97,11 @@ public OmSnapshot load(@Nonnull String snapshotTableKey)
         // see if the snapshot exists
         snapshotInfo = getSnapshotInfo(snapshotTableKey);
 
-        boolean isSnapshotInCache =
+        CacheValue<SnapshotInfo> cacheValue =
             ozoneManager.getMetadataManager().getSnapshotInfoTable()
-                .getCacheValue(new CacheKey<>(snapshotTableKey)) != null;
+                .getCacheValue(new CacheKey<>(snapshotTableKey));
+        boolean isSnapshotInCache = cacheValue != null && Optional.ofNullable(
+            cacheValue.getCacheValue()).isPresent();
```

Comment:
Verified manually that this approach would work:

<img width="1469" alt="image" src="https://user-images.githubusercontent.com/50227127/221980152-37acb87e-31ae-45e6-b7a7-4f897d0139d2.png">